src/util/Xhr.js

```diff
@@ -69,39 +81,90 @@ class Xhr {
         sharedLinkPassword,
         responseInterceptor,
         requestInterceptor,
+        shouldRetry = true,
     }: Options = {}) {
         this.id = id;
         this.token = token;
         this.clientName = clientName;
         this.version = version;
         this.sharedLink = sharedLink;
         this.sharedLinkPassword = sharedLinkPassword;
-        this.responseInterceptor = responseInterceptor;
+        this.responseInterceptor = responseInterceptor || this.defaultResponseInterceptor;
         this.axios = axios.create();
         this.axiosSource = axios.CancelToken.source();
+        this.retryCount = 0;
+        this.shouldRetry = shouldRetry;
 
-        if (typeof responseInterceptor === 'function') {
-            // Called on any non 2xx response
-            this.axios.interceptors.response.use(responseInterceptor, this.errorInterceptor);
-        }
+        this.axios.interceptors.response.use(this.responseInterceptor, this.errorInterceptor);
 
         if (typeof requestInterceptor === 'function') {
             this.axios.interceptors.request.use(requestInterceptor);
         }
     }
 
+    /**
+     * Default response interceptor which just returns the response
+     *
+     * @param {Object} response - the axios response
+     * @return the response
+     */
+    defaultResponseInterceptor(response: $AxiosXHR<any>) {
+        return response;
+    }
+
+    /**
+     * Determines if a request should be retried
+     *
+     * @param {Object} error - Error object from axios
+     * @return {boolean} true if the request should be retried
+     */
+    shouldRetryRequest(error: $AxiosError<any>): boolean {
+        if (!this.shouldRetry || this.retryCount >= MAX_NUM_RETRIES) {
+            return false;
+        }
+
+        const { response, request } = error;
+        // Retry if there is a network error (e.g. ECONNRESET) or rate limited
+        const isNetworkError = request && !response;
+        const isRetryableStatusCode = isNetworkError || getProp(response, 'status') === HTTP_STATUS_CODE_RATE_LIMIT;
```

Comment:
Obviously the more generic way of doing this would be to let the caller tell what HTTP codes to retry on. But if/when we get use cases of more, specially if some functions want to retry on different codes, then we can come back and refactor.

Reply:
Yeah, It should be easy to add in an optional option which implements a custom function but defaults.